pkg/types/spec/validations.go

```diff
@@ -817,28 +813,53 @@ func validatePredictor(
 func validateMultiModelsFields(api *userconfig.API) error {
 	predictor := api.Predictor
 
-	if len(predictor.Models.Paths) == 0 && predictor.Models.Dir == nil {
-		return errors.Wrap(ErrorSpecifyOneOrTheOther(userconfig.ModelsPathsKey, userconfig.ModelsDirKey), userconfig.ModelsKey)
+	var models *userconfig.MultiModels
+	if api.Predictor.Models != nil {
+		if api.Predictor.Type == userconfig.PythonPredictorType {
+			return ErrorFieldNotSupportedByPredictorType(userconfig.ModelsKey, api.Predictor.Type)
+		}
+		models = api.Predictor.Models
+	}
+	if api.Predictor.MultiModelReloading != nil {
+		if api.Predictor.Type != userconfig.PythonPredictorType {
+			return ErrorFieldNotSupportedByPredictorType(userconfig.MultiModelReloadingKey, api.Predictor.Type)
+		}
+		models = api.Predictor.MultiModelReloading
+	}
+	if models == nil && api.Predictor.Type != userconfig.PythonPredictorType {
+		return ErrorFieldMustBeDefinedForPredictorType(userconfig.ModelsKey, api.Predictor.Type)
+	}
+
+	if models.Path == nil && len(models.Paths) == 0 && models.Dir == nil {
```

Comment:
Would this panic for python predictor which does not specify models?

Reply:
Yes, this panics. It is now addressed.